src/win/core.c

```diff
@@ -318,6 +319,16 @@ static void uv__poll(uv_loop_t* loop, int timeout) {
          */
         if (overlappeds[i].lpOverlapped) {
           req = container_of(overlappeds[i].lpOverlapped, uv_req_t, u.io.overlapped);
+          /* If multiple async handles were triggered we might end up with
+           * multiple UV_WAKEUP requests (IOCP completion events). They all
+           * share the same req however, so we need to be careful to only make
```

Comment:
I don't understand the "they all share the same req" part. The object that is posted with `PostQueuedCompletionStatus()` is `&handle->async_req.u.io.overlapped`, isn't it?

That's unique for each handle and should (barring bugs) only be posted once until it's processed by `uv_process_reqs()`.

There might be concurrency bugs in our handling of `handle->async_sent` though. 37042a5b fixed one such bug in the unix implementation.

Reply:
@bnoordhuis There is one single async_req per loop: https://github.com/libuv/libuv/blob/7c845fb25f13a03a5264ea28e0a642fc7dbf7c98/src/win/async.c#L63

Reply:
Oh... it's per-handle in the v1.x branch but per-loop in master. Okay, objection withdrawn.